### PR TITLE
ease cpf adapter

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Tested manually on `gauss`.

File: service/controller/v24/resource/cpf/create.go [hunk 1]
```diff
@@ -127,3 +131,77 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		for _, cidrBlock := range tenantPrivateSubnetCidrs {
+			route := adapter.CPFRouteTablesRoute{
+				RouteTableName: name,
+				RouteTableID:   id,
+				// Requester CIDR block, we create the peering connection from the
+				// guest's private subnets.
```
Comment (rossf7) [inline service/controller/v24/resource/cpf/create.go]:
```suggestion
				// tenant's private subnets.
```

File: service/controller/v24/resource/cpf/create.go [hunk 2]
```diff
@@ -127,3 +131,77 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+		id, err := r.routeTable.IDForName(ctx, name)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		route := adapter.CPFRouteTablesRoute{
+			RouteTableName: name,
+			RouteTableID:   id,
+			// Requester CIDR block, we create the peering connection from the
+			// guest's CIDR for being able to access Vault's ELB.
```
Comment (rossf7) [inline service/controller/v24/resource/cpf/create.go]:
```suggestion
			// tenant's CIDR for being able to access Vault's ELB.
```

File: service/controller/v24/adapter/cpf.go [hunk 3]
```diff
@@ -1,25 +1,13 @@
...
-	AvailabilityZones          []v1alpha1.AWSConfigStatusAWSAvailabilityZone
 	BaseDomain                 string
 	ClusterID                  string
-	EncrypterBackend           string
 	GuestHostedZoneNameServers string
-	NetworkCIDR                string
+	PrivateRoutes              []CPFRouteTablesRoute
+	PublicRoutes               []CPFRouteTablesRoute
 	Route53Enabled             bool
 }
```
Comment (kopiczko) [inline service/controller/v24/adapter/cpf.go]: Do we even need to use `*Config`? Can't we just name it `CPF` and use it directly as a pure data struct?

**Comment by xh3b4sd** [comment]
I keep the refactoring and renaming up. Step by step.

File: service/controller/v24/adapter/cpf.go (hunk 3 above)
Comment (xh3b4sd) [reply service/controller/v24/adapter/cpf.go]: I am planning to do this. I also want to move the data structure to the templates. Will do with the next PRs.

**Pull request merged**